### PR TITLE
Add log in job attempt api

### LINKED ISSUE #3967: add logger in job history api

### PR DESCRIPTION
**Pull request by debuggy**
closes #3967

### CODE REVIEW

File: src/rest-server/src/models/v2/job-attempt.js [hunk 1]
```diff
@@ -60,9 +61,11 @@ const healthCheck = async () => {
       if (result.statusCode === 200) {
         return true;
       } else {
+        logger.warn(`elastic search return status: ${result.statusCode}`);
```
Comment (yqwang-ms) [inline src/rest-server/src/models/v2/job-attempt.js]: Also log http error msg?

File: src/rest-server/src/models/v2/job-attempt.js [hunk 2]
```diff
@@ -60,9 +61,11 @@ const healthCheck = async () => {
       if (result.statusCode === 200) {
         return true;
```
Comment (yqwang-ms) [inline src/rest-server/src/models/v2/job-attempt.js]: ignore index not found?

File: src/rest-server/src/models/v2/job-attempt.js [hunk 3]
```diff
@@ -102,6 +106,7 @@ const list = async (frameworkName) => {
       isLatest: true,
     });
   } else if (response.status === 404) {
+    logger.warn(`could not get framework ${uid} from k8s`);
     return {status: 404, data: null};
```
Comment (yqwang-ms) [inline src/rest-server/src/models/v2/job-attempt.js]: log err code and msg

File: src/rest-server/src/models/v2/job-attempt.js [hunk 4]
```diff
@@ -205,6 +211,7 @@ const get = async (frameworkName, jobAttemptIndex) => {
     uid = response.data.metadata.uid;
     attemptFramework = response.data;
   } else if (response.status === 404) {
+    logger.warn(`could not find framework: ${uid}`);
     return {status: 404, data: null};
```
Comment (yqwang-ms) [inline src/rest-server/src/models/v2/job-attempt.js]: log err code and msg

File: src/rest-server/src/models/v2/job-attempt.js (hunk 1 above)
Comment (debuggy) [reply src/rest-server/src/models/v2/job-attempt.js]: there is no message property in elastic search response, maybe status code is enough as a hint

File: src/rest-server/src/models/v2/job-attempt.js (hunk 2 above)
Comment (debuggy) [reply src/rest-server/src/models/v2/job-attempt.js]: could you explain in more details

File: src/rest-server/src/models/v2/job-attempt.js (hunk 4 above)
Comment (debuggy) [reply src/rest-server/src/models/v2/job-attempt.js]: done

File: src/rest-server/src/models/v2/job-attempt.js (hunk 3 above)
Comment (debuggy) [reply src/rest-server/src/models/v2/job-attempt.js]: done

File: src/rest-server/src/models/v2/job-attempt.js (hunk 1 above)
Comment (yqwang-ms) [reply src/rest-server/src/models/v2/job-attempt.js]: But we previously get "index not found" msg, how to get it?

File: src/rest-server/src/models/v2/job-attempt.js (hunk 2 above)
Comment (yqwang-ms) [reply src/rest-server/src/models/v2/job-attempt.js]: If "index not found" response, we also treat it as health.
Another apporach is not checking "framework" in es, you should just ping es.

File: src/rest-server/src/models/v2/job-attempt.js (hunk 1 above)
Comment (yqwang-ms) [reply src/rest-server/src/models/v2/job-attempt.js]: Log whole result (and response in other places)

File: src/rest-server/src/models/v2/job-attempt.js (hunk 2 above)
Comment (yqwang-ms) [reply src/rest-server/src/models/v2/job-attempt.js]: Check this https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-health.html

File: src/rest-server/src/models/v2/job-attempt.js [hunk 5]
```diff
@@ -54,15 +55,15 @@ const healthCheck = async () => {
     return false;
   } else {
     try {
-      const result = await elasticSearchClient.indices.get({
-        index: 'framework',
-      });
+      const result = await elasticSearchClient.cat.health();
       if (result.statusCode === 200) {
         return true;
```
Comment (yqwang-ms) [inline src/rest-server/src/models/v2/job-attempt.js]: Do we also need to check "status" field in the body?

Comment (debuggy) [reply src/rest-server/src/models/v2/job-attempt.js]: how to process with different status

**Pull request merged**